openhands/runtime/impl/remote/remote_runtime.py

```diff
@@ -377,7 +377,22 @@ def _wait_until_alive_impl(self):
         raise AgentRuntimeNotReadyError()
 
     def close(self):
-        if self.config.sandbox.keep_runtime_alive or self.attach_to_existing:
+        if self.attach_to_existing:
+            super().close()
+            return
+        if self.config.sandbox.keep_runtime_alive:
+            if self.config.sandbox.pause_closed_runtimes:
+                try:
+                    if not self._runtime_closed:
+                        with self._send_runtime_api_request(
+                            'POST',
+                            f'{self.config.sandbox.remote_runtime_api_url}/pause',
+                            json={'runtime_id': self.runtime_id},
+                        ):
+                            self.log('debug', 'Runtime paused.')
+                except Exception as e:
+                    self.log('error', f'Unable to pause runtime: {str(e)}')
+                    raise e
             super().close()
```

Comment:
It seems `super().close()` is still called after pausing the runtime? I'm not too familiar with this

Reply:
Yup that's intentional--super().close() just closes an http request session